src/agentscope/agents/dialog_agent.py

```diff
@@ -68,13 +68,13 @@ def reply(self, x: dict = None) -> dict:
             response to the user's input.
         """
         # record the input if needed
-        if x is not None:
+        if not self.memory:
```

Comment:
Should it be `if self.memory`? BTW, if use_memory is False, does `self.memory` defined?

Reply:
Thanks! The error is fixed